src/containers/Nodes/Nodes.tsx

```diff
@@ -58,27 +72,88 @@ export const Nodes = ({path, type, additionalNodesInfo = {}}: NodesProps) => {
-        useTypedSelector((state) => state.nodes);
+    const {
+        wasLoaded,
+        loading,
+        error,
+        nodesUptimeFilter,
+        searchValue,
+        sortOrder = ASCENDING,
+        sortValue = 'NodeId',
+        totalNodes,
+    } = useTypedSelector((state) => state.nodes);
     const problemFilter = useTypedSelector((state) => state.settings.problemFilter);
     const {autorefresh} = useTypedSelector((state) => state.schema);
 
     const nodes = useTypedSelector(selectFilteredNodes);
 
     const [useNodesEndpoint] = useSetting(USE_NODES_ENDPOINT_IN_DIAGNOSTICS_KEY);
+    const [useBackendParamsForTables] = useSetting<boolean>(USE_BACKEND_PARAMS_FOR_TABLES_KEY);
 
-    const fetchNodes = useCallback(() => {
-        // For not DB entities we always use /compute endpoint instead of /nodes
-        // since /nodes can return data only for tenants
-        if (path && (!useNodesEndpoint || !isDatabaseEntityType(type))) {
-            dispatch(getComputeNodes({path}));
-        } else {
-            dispatch(getNodes({tenant: path}));
+    const [requestParams, setRequestParams] = useState<NodesGeneralRequestParams>({});
+
+    // If backend params are enabled, update params value and refetch data on filters change
+    // Otherwise no params will be sent and data will be filtered inside selector
+    useEffect(() => {
+        if (useBackendParamsForTables) {
+            const problemsOnly = problemFilter === ProblemFilterValues.PROBLEMS;
+            const uptime =
+                nodesUptimeFilter === NodesUptimeFilterValues.SmallUptime
+                    ? HOUR_IN_SECONDS
+                    : undefined;
+
+            setRequestParams({
+                filter: searchValue,
+                problems_only: problemsOnly,
+                uptime,
+                sortOrder,
+                sortValue,
+            });
         }
-    }, [dispatch, path, type, useNodesEndpoint]);
+    }, [
+        useBackendParamsForTables,
+        searchValue,
+        problemFilter,
+        nodesUptimeFilter,
+        sortOrder,
+        sortValue,
+    ]);
+
+    const fetchNodes = useCallback(
+        (isBackground) => {
+            if (!isBackground) {
+                dispatch(setDataWasNotLoaded());
+            }
+
+            // For not DB entities we always use /compute endpoint instead of /nodes
+            // since /nodes can return data only for tenants
+            if (path && (!useNodesEndpoint || !isDatabaseEntityType(type))) {
+                dispatch(getComputeNodes({path, ...requestParams}));
+            } else {
+                dispatch(getNodes({tenant: path, ...requestParams}));
+            }
+        },
+        [dispatch, path, type, useNodesEndpoint, requestParams],
+    );
 
     useAutofetcher(fetchNodes, [fetchNodes], isClusterNodes ? true : autorefresh);
 
+    const sort: SortOrder | undefined = useMemo(() => {
```

Comment:
If we are going to use different sort types in other places, it may be useful to move calculating sort params (and maybe even `handleSort`) into a hook or a helper.